service/lib/agama/software/product_builder.rb

```diff
@@ -64,6 +64,7 @@ def initialize_product(id, data, attrs)
           product.description = attrs["description"]
           product.name = data[:name]
           product.version = data[:version]
+          product.icon = attrs["icon"] if attrs.key?("icon") && !attrs["icon"].nil?
```

Comment:
default value for missing key in ruby is nil so check for nil is enough...and also nil is evaluated to false so you can use something like
```suggestion
          product.icon = attrs["icon"] if attrs["icon"]
```

another option is to define default icon as constant like `DEFAULT_ICON` and then use here
```suggestion
          product.icon = attrs["icon"] || DEFAULT_ICON
```